healvis/sky_model.py

```diff
@@ -11,12 +11,19 @@
 import os
 import h5py
 from astropy.cosmology import Planck15 as cosmo
+import healpy as hp
 
 from .utils import mparray, comoving_voxel_volume, comoving_distance
 
-# Moving a lot from eorsky to here.
+try:
+    import pygsm
+    pygsm_import = True
+except ImportError:
+    raise Warning("pygsm package not found. This is required to use GSM functionality.")
```

Comment:
This should use the warnings module, otherwise this will stop execution if `pygsm` is missing.

The warning should be moved down to the `gsm_shell` function. There's no reason to warn the user about a missing optional dependency unless they try to use it.